model/src/application/command/option.rs

```diff
@@ -408,15 +432,40 @@ impl CommandOptionType {
             CommandOptionType::Channel => "Channel",
             CommandOptionType::Role => "Role",
             CommandOptionType::Mentionable => "Mentionable",
+            CommandOptionType::Number => "Number",
         }
     }
 }
 
+#[derive(Clone, Copy, Debug, Deserialize, Serialize)]
+pub struct Number(f64);
```

Comment:
Can this also implement From<f64> to allow us to use `val.into()` when creating float types?